src/test/java/org/jenkinsci/plugins/casc/PluginManagerConfiguratorTest.java

```diff
@@ -24,4 +31,51 @@ public void testInstallPlugins() {
         assertNotNull(chucknorris);
         assertEquals("1.0", chucknorris.getWrapper().getVersion());
     }
+
+    @Test
+    public void describeDefaultConfig() throws Exception {
+        final PluginManagerConfigurator root = getPluginManagerConfigurator();
+        final CNode node = root.describe(root.getTargetComponent());
+        assertNotNull(node);
+        assertTrue(node instanceof Mapping);
+        final Object sites = ((Mapping) node).get("sites");
+        assertNotNull(sites);
+        assertTrue(sites instanceof Sequence);
+        assertEquals(1, ((Sequence) sites).size());
+        final Object site = ((Sequence) sites).get(0);
+        assertNotNull(site);
+        assertTrue(site instanceof Mapping);
+        assertEquals("default", ((Mapping) site).get("id").toString());
+        assertTrue(((Mapping) site).containsKey("url"));
+    }
+
+    @Test
+    @ConfiguredWithCode("ProxyConfigTest.yml")
+    public void describeProxyConfig() throws Exception {
+        final PluginManagerConfigurator root = getPluginManagerConfigurator();
+        final CNode configNode = root.describe(root.getTargetComponent());
+        ((Mapping) configNode).remove("sites");
+        final String yamlConfig = toYamlString(configNode);
+        assertEquals(String.join("\n",
+                "proxy:",
```

Comment:
I don't think we can guarantee the attribute ordering exported in yaml, can we ? Assert on CNode is more predictible afaik

Reply:
Experimentally, it is always alphabetically sorted.
Is that OK ? I can try to find the relevant `snakeyaml` code or spec that explictely ensure this.
IHMO the benefit in terms of clarity for the tests is worth it.

Reply:
this is not snakeYaml but `toYaml` which translates exported `CNode`s into yaml. snakeyaml mapping's tuples are ordered, but I don't think we can guarantee ordering for the Mapping CNode (based on HashMap.entrySet), but relying on implementation details (maybe implementation specific ?).
Or maybe we could have `toYaml` sort tuples by key name to become fully deterministic.

Reply:
i.e in org.jenkinsci.plugins.casc.ConfigurationAsCode#toYaml use
```java
final List<Map.Entry<String, CNode>> entries = new ArrayList<>(mapping.entrySet());
                Collections.sort(entries, new Comparator<Map.Entry<String, CNode>>() {
                    @Override
                    public int compare(Map.Entry<String, CNode> o1, Map.Entry<String, CNode> o2) {
                        return o1.getKey().compareTo(o2.getKey());
                    }
                });
                for (Map.Entry<String, CNode> entry : entries) {
```

Reply:
OK, I see.
Thanks for the explanation and code snippet !
I made the change, ponly slightly shorter thanks to IntelliJ advices :)